egs2/ami/asr1/local/data.sh

```diff
@@ -0,0 +1,94 @@
+#       the 8 source microphones].
+# ... by calling run.sh as, for example,
+# ./run.sh --mic sdm1
+# ./run.sh --mic mdm8
+mic=ihm
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+if [ ! -e "${AMI}" ]; then
+    log "Fill the value of 'AMI' of db.sh"
+    exit 1
+fi
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+base_mic=${mic//[0-9]/} # sdm, ihm or mdm
+nmics=${mic//[a-z]/} # e.g. 8 for mdm8.
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "data stage 1: Data Download"
+    if [ -d ${AMI} ] && ! touch ${AMI}/.foo 2>/dev/null; then
+	log "$0: directory $AMI seems to exist and not be owned by you."
+	log " ... Assuming the data does not need to be downloaded.  Please use --stage 2."
+	exit 1
+    fi
+    if [ -e data/local/downloads/wget_${mic}.sh ]; then
+	log "data/local/downloads/wget_$mic.sh already exists, better quit than re-download... (use --stage N)"
+	exit 1
+    fi
+    local/ami_download.sh ${mic} ${AMI}
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    log "data stage 2: Data preparation"
+
+    # common data prep
+    if [ ! -d data/local/downloads ]; then
+	local/ami_text_prep.sh data/local/downloads
+    fi
+
+    # beamforming
+    if [ "$base_mic" == "mdm" ]; then
+	PROCESSED_AMI_DIR=${PWD}/beamformed
+	if [ -z ${BEAMFORMIT} ]; then
+	    export BEAMFORMIT=${KALDI_ROOT}/tools/BeamformIt
+	fi
+	export PATH=${PATH}:${BEAMFORMIT}
+	! hash BeamformIt && log "Missing BeamformIt, run 'cd ../../../tools/kaldi/tools; extras/install_beamformit.sh; cd -;'" && exit 1
+	local/ami_beamform.sh --cmd "${train_cmd}" --nj 20 ${nmics} ${AMI} ${PROCESSED_AMI_DIR}
+    else
+	PROCESSED_AMI_DIR=${AMI}
+    fi
+    local/ami_${base_mic}_data_prep.sh ${PROCESSED_AMI_DIR} ${mic}
+    # data augmentation
+    utils/perturb_data_dir_speed.sh 0.9 data/${mic}/train_orig data/ihm_tmp1
+    utils/perturb_data_dir_speed.sh 1.0 data/${mic}/train_orig data/ihm_tmp2    
+    utils/perturb_data_dir_speed.sh 1.1 data/${mic}/train_orig data/ihm_tmp3
+    rm -r data/${mic}/train_orig
+    utils/combine_data.sh --extra-files utt2uniq data/ihm/train_orig data/ihm_tmp1 data/ihm_tmp2 data/ihm_tmp3
+    
+    local/ami_${base_mic}_scoring_data_prep.sh ${PROCESSED_AMI_DIR} ${mic} dev
+    local/ami_${base_mic}_scoring_data_prep.sh ${PROCESSED_AMI_DIR} ${mic} eval
+    for dset in train dev eval; do
+	# changed the original AMI data structure in the Kaldi recipe to the following
+	utils/data/modify_speaker_info.sh --seconds-per-spk-max 30 data/${mic}/${dset}_orig data/${mic}_${dset}
+    done
```

Comment:
Remove `utils/data/modify_speaker_info.sh` because we can't use kaldi command by default in espnet2,but modify_speaker_info requires  `wav-to-duration`. We don't need to modify speaker.

Reply:
do you mean we cannot use any utils scripts from kaldi?

Reply:
I mean we can't use binary by default and we still use utils script and we can also still use fbank_pitch just as an option.
modify_speaker_info.sh invokes wav-to-duration and it's c++ command, so I'd like to avoid it.